### ISSUE TITLE
s1-helper logs showing tls error

### ISSUE THREAD

**Issue by Ujjawal-Gupta** [Sentinel-One/helm-charts#164]
I have used the helm to install the s1 agent in a k8s cluster but the s1-helper logs are showing:
 TLS handshake error from 10.x.x.x:xxxx: remote error: tls: unknown certificate authority.

**Comment by kdeyko** [comment]
We have the same issue.
It appears to happen when the `s1-agent-helper` pod gets recreated _after_ the `s1-agent` pods (that is, when helper is older than the agents). Recreating the agent pods solves the issue (until the next time we reboot helper pod for some reason).

The possible plausible cases for just helper reboots:
- pod eviction
- requests/limits changes

**Comment by oded-s1** [comment]
@kdeyko @pratik-pagade can you please provide more info on how to recreate the issue?

**Comment by kdeyko** [comment]
There are no clear steps to reproduce the issue, as it seem to appear quite randomly.
Here's our setup:
- k8s v1.29 on AWS EKS
- [This](https://github.com/Sentinel-One/helm-charts/tree/master/charts/s1-agent) very Helm chart

Values:
```yaml
configuration:
  repositories:
    agent: "our-cool-registry/s1agent"
    helper: "our-cool-registry/s1helper"
  tag:
    agent: "24.1.2-x86_64"
    helper: "24.1.2-x86_64"
  env:
    agent:
      log_level: "error"
    helper:
      log_level: "error"
agent:
  resources:
    limits:
      cpu: null
      memory: 512Mi
    requests:
      cpu: 100m
      memory: 512Mi
helper:
  resources:
    limits:
      cpu: null
      memory: 128Mi
    requests:
      cpu: 10m
      memory: 128Mi
```

Again, this may or may not work for you, but I once saw the issue appear after we tweaked the `resources` values for agent or helper and then re-applied the chart. It also happened during the k8s version upgrade, when all the nodes were recreated, thus all the s1 pods were recreated too.

**Comment by kdeyko** [comment]
It looks like the issue with TLS certificate was mitigated using [this technique](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments) in [this commit](https://github.com/Sentinel-One/helm-charts/commit/f4bcfc887d2988b35fff66908f8aed035c7202c7). This patch was included in v24.3.2 of the chart.
I tried it, it works. There was a burst of those TLS errors during the rolling restart, but it settled down after that. I'll consider this solution as good enough.

The root issue lies [here](https://github.com/Sentinel-One/helm-charts/blob/master/charts/s1-agent/templates/_helpers.tpl#L202): the chart generates new CA on _each_ Helm upgrade, even if there were no changes to agent or helper pods. So the related secrets are changed, but the pods were never restarted to see it.
The workaround mentioned above forces rolling restart of the pods on each Helm upgrade operation.
